frontend/javascripts/oxalis/controller/combinations/skeletontracing_plane_controller.js

```diff
@@ -144,6 +146,113 @@ export function moveNode(dx: number, dy: number) {
   );
 }
 
+function otherNodeOfEdge(edge: Edge, nodeId: number): number {
+  return edge.source === nodeId ? edge.target : edge.source;
+}
+function getSubsequentNodeFromTree(tree: Tree, node: Node, excludedId: ?number): number {
+  const nodes = tree.edges
+    .getEdgesForNode(node.id)
+    .map(edge => otherNodeOfEdge(edge, node.id))
+    .filter(id => id !== excludedId);
+  const next = nodes.length ? Math.max(...nodes) : node.id;
+  return next;
+}
+function getPrecedingNodeFromTree(tree: Tree, node: Node, excludedId: ?number): number {
+  const nodes = tree.edges
+    .getEdgesForNode(node.id)
+    .map(edge => otherNodeOfEdge(edge, node.id))
+    .filter(id => id !== excludedId);
+  const prev = nodes.length ? Math.min(...nodes) : node.id;
+  return prev;
+}
+
+function toSubsequentNode(): void {
+  const tracing = enforceSkeletonTracing(Store.getState().tracing);
+  const { navigationList, activeNodeId, activeTreeId } = tracing;
+  let isValidList = true;
+  if (activeNodeId == null || activeNodeId === undefined) return;
+
+  if (
+    activeNodeId !== navigationList.list[navigationList.activeIndex] ||
+    !navigationList.list.length
+  ) {
+    isValidList = false;
+  }
+
+  if (
+    navigationList.list.length > 1 &&
+    navigationList.activeIndex < navigationList.list.length - 1 &&
+    isValidList
+  ) {
+    // navigate to subsequent node in list
+    Store.dispatch(setActiveNodeAction(navigationList.list[navigationList.activeIndex + 1]));
+    Store.dispatch(updateNavigationListAction(navigationList.list, navigationList.activeIndex + 1));
+  } else {
+    // search for subsequent node in tree
+    const { tree, node } = getNodeAndTree(tracing, activeNodeId, activeTreeId)
+      .map(([maybeTree, maybeNode]) => ({ tree: maybeTree, node: maybeNode }))
+      .getOrElse({
+        tree: null,
+        node: null,
+      });
```

Comment:
Maybe simplify this to

```javascript
const [ tree, node ] = getNodeAndTree(tracing, activeNodeId, activeTreeId)
  .getOrElse([null, null]);
```
Same in the other method.

Reply:
Flow is not happy with that :( . Maybe I could extract it in an extra method so it at least doesn't appear twice?

Reply:
:/ Yes you could put it into a `getNodeAndTreeOrNull` method in the skeleton tracing accessor file or you could annotate my proposed solution with a `// $FlowFixMe` to silence flow (as we know that the code is correct).